File: solutions/go/largest-series-product/1/largest_series_product.go
// This package determines the largest product for a set sequences
package lsproduct

import ("strconv"
"errors"
)

// This function determines largest product for a certain size
// of numbers
func LargestSeriesProduct(digits string, span int) (int64, error) {
	var mp int64 = 0
	if span > len(digits) {
		return mp, errors.New("span greater than len of digits")
	}
	if len(digits)> 1 && span < 0 {
		return mp, errors.New("span less than 1")
	}
	for di := 0; di+span<=len(digits); di++ {
		var p int64 = 1
		for i := 0; i<span; i++ {
			v,err := strconv.Atoi(string(digits[di+i:di+i+1]))
			if err != nil {
				return 0,errors.New("Bad digit")
			}
			p *= int64(v)
		}
		if p>mp {mp=p}
	}
	return mp, nil
}

File: solutions/c/largest-series-product/1/largest_series_product.c
#include "largest_series_product.h"
#include <string.h>
#include <stdio.h>
#include <ctype.h>

int64_t largest_series_product(char *digits, size_t span){
    size_t dl=strlen(digits);
    int64_t lp=0;
    if (span>dl) return -1;
    for (size_t di=0; di<dl-span+1; di++){
        int64_t p=1;
        for (size_t si=0; si<span; si++) {
            if (!isdigit(digits[di+si])) return -1;
            p*=(int)(digits[di+si]-'0'); 
        }
        if (p>lp) lp=p;
    }
    return lp;
}